### PR TITLE
Fix test_events flakiness

### PR DESCRIPTION
**Pull request by zbud-msft**
### Description of PR

Summary:
Fixes # (issue) 28728116

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?

This PR will

1) Add test_events back to PR checker
2) Fix event_down container test flakiness
3) Remove unneeded test_events_cache test as our tests already use cache for testing

#### How did you do it?

Code change

#### How did you verify/test it?

Lab test and Pipeline

### CODE REVIEW

**Comment by wsycqyz** [comment]
Do we need this PR in other branch?

File: tests/telemetry/conftest.py [hunk 1]
```diff
@@ -175,6 +175,9 @@ def test_eventd_healthy(duthosts, enum_rand_one_per_hwsku_hostname, ptfhost, set
     if succeeded and ('eventd' not in features_dict or features_dict['eventd'] == 'disabled'):
         pytest.skip("eventd is disabled on the system")
 
+    if duthost.is_multi_asic:
```
Comment (qiluo-msft) [inline tests/telemetry/conftest.py:178]: >is_multi_asic [](http://example.com/codeflow?start=15&length=13)

Let's use a separate PR for this bugfix relating multi-asic.

File: tests/telemetry/telemetry_utils.py [hunk 2]
```diff
@@ -139,7 +130,8 @@ def trigger_logger(duthost, log, process, container="", priority="local0.notice"
 
 def generate_client_cli(duthost, gnxi_path, method=METHOD_GET, xpath="COUNTERS/Ethernet0", target="COUNTERS_DB",
                         subscribe_mode=SUBSCRIBE_MODE_STREAM, submode=SUBMODE_SAMPLE,
-                        intervalms=0, update_count=3, create_connections=1, filter_event_regex=""):
+                        intervalms=0, update_count=3, create_connections=1, filter_event_regex="",
+                        timeout=-1):
```
Comment (qiluo-msft) [inline tests/telemetry/telemetry_utils.py:134]: >timeout [](http://example.com/codeflow?start=24&length=7)

Could you add some function comments to explain each parameter? Especial for this one, what is the contract? What is the timeout units?

Comment (zbud-msft) [reply tests/telemetry/telemetry_utils.py:134]: Added comments

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**